src/App.js

```diff
@@ -1,10 +1,36 @@
+/* eslint-disable no-console */
 import React from 'react';
 
 import './App.css';
 
 import posts from './api/posts';
 import comments from './api/comments';
 import users from './api/users';
+import PostsList from './components/PostsList/PostsList';
+
+function findAuthor(userId) {
+  const person = users.filter(user => user.id === userId);
+  const address = Object.entries(person[0].address)
+    .slice(0, 4)
+    .map(option => option.join(' : '))
+    .join(', ');
+
+  return [person[0].name, person[0].email, address];
+}
+
+function findComments(id) {
+  return comments.filter(comment => comment.postId === id);
+}
+
+const prepearedPosts = posts.map(post => ({
+  ...post,
+  author: findAuthor(post.userId)[0],
+  email: findAuthor(post.userId)[1],
+  address: findAuthor(post.userId)[2],
+  comments: findComments(post.id),
+}));
```

Comment:
```suggestion
const prepearedPosts = posts.map(post => {
const [author, email, address] = findAuthor(post.userId);
return {
  ...post,
  author,
  email,
  address,
  comments: findComments(post.id),
}});
```
Зачем трижды запускать одну и ту же функцию для одного элемента, если достаточно запустить ее один раз?